tests/components/http/test_forwarded.py

```diff
@@ -0,0 +1,487 @@
+"""Test real forwarded middleware."""
+from ipaddress import ip_network
+
+from aiohttp import web
+from aiohttp.hdrs import X_FORWARDED_FOR, X_FORWARDED_HOST, X_FORWARDED_PROTO
+import pytest
+
+from homeassistant.components.http.forwarded import async_setup_forwarded
+
+
+async def mock_handler(request):
+    """Return the real IP as text."""
+    return web.Response(text=request.remote)
+
+
+async def test_x_forwarded_for_without_trusted_proxy(aiohttp_client, caplog):
+    """Test that we get the IP from the transport."""
+
+    async def handler(request):
+        url = mock_api_client.make_url("/")
+        assert request.host == f"{url.host}:{url.port}"
+        assert request.scheme == "http"
```

Comment:
Codacy found an issue: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5985897)